### PR TITLE
`Assessment`: Improve feedback deletion confirmation alert

### PR DESCRIPTION
**Pull request by Strohgelaender**
#### General
- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [ ] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
Deleting a feedback triggers a browser confirmation alert. We usually don't use such confirmation alerts and instead use our own delete confirmation dialog.

### Description
This PR replaces the confirmation alert with the typical delete dialog. Instead of the feedback id (which provides no value to the tutor) the detail text is shown in the confirmation alert.

### Steps for Testing
Prerequisites:
- 1 Tutor
- 1 Programming Exercise

1. Assess the submission and add feedback (inline feedback and general feedback)
2. Delete the feedback again
3. The confirmation alert looks good.

Note: Feedback for modeling elements and text blocks use a different deletion mechanism, this PR only changes general feedback and code inline feedback.

#### Performance Review
- [x] I confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots
old: 
grafik

now: 
grafik

### CODE REVIEW

**Review by MichaelOwenDyer — APPROVED** [review]
Looks good, but I question whether it is wise to include the entire feedback text in the confirmation window... maybe just a snippet would be better?

File: src/main/webapp/i18n/en/feedback.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
             "updated": "Updated Feedback with identifier {{ param }}",
             "deleted": "Deleted Feedback with identifier {{ param }}",
             "delete": {
-                "question": "Are you sure you want to delete Feedback {{ id }}?"
+                "question": "Are you sure you want to delete the Feedback \"{{ text }}\"?"
```
Comment (JohannesStoehr) [inline src/main/webapp/i18n/en/feedback.json:12]:
```suggestion
                "question": "Are you sure you want to delete the feedback \"{{ text }}\"?"
```
I'm also thinking about adding a colon, what is your opinion?
```suggestion
                "question": "Are you sure you want to delete the feedback: \"{{ text }}\"?"
```

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]
Code looks good overall. Left one nitpick:

**Review by undernagruzez — APPROVED** [review]
Tested on ts5, works as expected

**Review by bassner — CHANGES_REQUESTED** [review]
Looks like we have a small merge conflict here

**Review by JohannesStoehr — DISMISSED** [review]
Approve code

**Review by RY997 — DISMISSED** [review]
Tested on ts4, worked well

**Review by MaximilianAnzinger — APPROVED** [review]
Code LGTM

**Review by DominikRemo — APPROVED** [review]
Code LGTM

File: src/main/webapp/app/assessment/assessment-detail/assessment-detail.component.ts [hunk 2]
```diff
@@ -16,17 +17,21 @@ export class AssessmentDetailComponent {
...
     faRobot = faRobot;
     faQuestionCircle = faQuestionCircle;
     faExclamation = faExclamation;
     faExclamationTriangle = faExclamationTriangle;
 
-    constructor(
-        private translateService: TranslateService,
-        public structuredGradingCriterionService: StructuredGradingCriterionService,
+
+        private dialogErrorSource = new Subject<string>();
```
Comment (terlan98) [inline src/main/webapp/app/assessment/assessment-detail/assessment-detail.component.ts:31]:
```suggestion
    private dialogErrorSource = new Subject<string>();
```

**Review by terlan98 — CHANGES_REQUESTED** [review]
Tested on TS5. The new alert works as described. I found 2 issues:
- I found 1 indentation issue, which is also causing the client tests to fail (because to Prettier is complaining)
- I'm not sure if it's related to this PR, but when the user opens an ongoing assessment and deletes an existing feedback, the save and submit buttons do not get enabled. The user has to add a new feedback or edit an existing one to enable the buttons.

**Review by terlan98 — APPROVED** [review]
Thanks for addressing my comment. Approved 👍

**Review by RY997 — APPROVED** [review]
Tested on ts1, worked well. Reapprove!

**Comment by krusche** [comment]
Will merge as soon as the checks are executed successfully

**Pull request merged**